### ISSUE TITLE
possible indentation bug

### ISSUE THREAD

**Issue by phlummox** [stackbuilders/stache#44]
I believe the following result is contrary to the spec, though I could be mistaken.

If we define 3 templates:

```
inner   = "xxx"
middle  = "{{>inner}}"
outer   = "{{>middle}}"
```
and have all of them in the template cache with appropriate names, then I believe the result of rendering `middle` should be exactly the same as the result of rendering `outer`. However, `stache` seems to give different results for the two.

**Steps to reproduce:**

-   `git clone` from 05e3c2a, `cd` into the repo
-   compile and run the following program:

    ```haskell
    {-# LANGUAGE OverloadedStrings #-}
    
    module Main
      where
    
    import Text.Mustache
    import Data.Text ( Text )
    import Data.Aeson
    
    default (Text)
    
    inner   = "xxx"
    middle  = "{{>inner}}"
    outer   = "{{>middle}}"
    
    main :: IO ()
    main = do
      let inner'  = either (error . show) id $
                      compileMustacheText "inner" inner
          middle' = either (error . show) id $
                      compileMustacheText "middle" middle
          outer'  = either (error . show) id $
                      compileMustacheText "outer" outer
    
          res1 = renderMustache (middle' <> inner')           (toJSON ())
          res2 = renderMustache (outer' <> middle' <> inner') (toJSON ())
    
      print $ res1 == res2
      ```
      (I compiled with stack, using `stack --resolver=lts-14 init; stack build; stack ghc -- --make possible_bug.hs; ./possible_bug`)

**Expected result:**

-  If my understanding of the spec is correct, the program should print `True`.

**Actual result:**

-  The program prints `False`; res2 is indented by 1 space.

The same behaviour exists in stache version 1.2.1.

I compared stache's behaviour with a Python implementation which claims to be spec-compliant, [chevron](https://github.com/noahmorrison/chevron); the equivalent Python program seems to give the behaviour I would expect. The following function returns `True`:

```python
#!/usr/bin/env python3

import chevron

def test():
  inner   = "xxx"
  middle  = "{{>inner}}"
  outer   = "{{>middle}}"

  def render(tpl):
    return chevron.render(template=tpl, data={},
                          partials_dict={
                              "inner" : inner,
                              "middle" : middle
                          } )

  return render(outer) == render(middle)
```

**Comment by mrkkrp** [comment]
This may be a bug. Although we test against the official spec (and so formally the implementation is conforming to the spec), it may be incomplete. Would you like to try to prepare a PR?

**Comment by phlummox** [comment]
No worries. Yes, I see now the tests in the `mustache-spec` directory that test against the spec; but you are right that the spec looks very incomplete. (mustache/spec#89 looks like it would add a test to the spec to cover this case, but it has been sitting unaddressed since August 2015.)

At any rate, I'll prepare a PR and let you decide. The behaviour results from [`addIndents`](https://github.com/stackbuilders/stache/blob/master/Text/Mustache/Render.hs#L253), which adds a megaparsec `Pos 1` to another `Pos 1` and gets `Pos 2` ... but `Pos 1` represents offset *zero* in a file, so really, the result should be `Pos 1` as well.

**Comment by mrkkrp** [comment]
Released 2.1.1 with the fix.
